crates/uv-tool/src/lib.rs

```diff
@@ -374,6 +374,47 @@ pub fn find_executable_directory() -> Result<PathBuf, Error> {
         .ok_or(Error::NoExecutableDirectory)
 }
 
+/// Find a directory to place manpages in.
+///
+/// This follows, in order:
+///
+/// - `$UV_TOOL_BIN_DIR/../share/man` (only choose this if `$UV_TOOL_BIN_DIR/../share` exists)
+/// - `$XDG_BIN_HOME/../share/man` (only choose this if `$XDG_BIN_HOME/../share` exists)
+/// - `$XDG_DATA_HOME/man`
+/// - `$HOME/.local/share/man`
+///
+/// On all platforms.
+///
+/// Errors if a directory cannot be found.
+pub fn find_manpage_directory() -> Result<PathBuf, Error> {
+    std::env::var_os("UV_TOOL_BIN_DIR")
+        .and_then(dirs_sys::is_absolute_path)
+        .map(|path| path.join("..").join("share").join("man"))
+        .and_then(|path| normalize_absolute_path(&path).ok())
+        .filter(|path| path.is_dir())
+        .or_else(|| {
+            std::env::var_os("XDG_BIN_HOME")
+                .and_then(dirs_sys::is_absolute_path)
+                .map(|path| path.join("..").join("share").join("man"))
+                .and_then(|path| normalize_absolute_path(&path).ok())
+                .filter(|path| path.is_dir())
+        })
+        .or_else(|| {
+            std::env::var_os("XDG_DATA_HOME")
+                .and_then(dirs_sys::is_absolute_path)
+                .map(|path| path.join("man"))
+        })
+        .or_else(|| {
+            // See https://github.com/dirs-dev/dirs-rs/blob/50b50f31f3363b7656e5e63b3fa1060217cbc844/src/win.rs#L5C58-L5C78
+            #[cfg(windows)]
+            let home_dir = dirs_sys::known_folder_profile();
+            #[cfg(not(windows))]
+            let home_dir = dirs_sys::home_dir();
+            home_dir.map(|path| path.join(".local").join("share").join("man"))
+        })
+        .ok_or(Error::NoExecutableDirectory)
+}
```

Comment:
According to https://man7.org/linux/man-pages/man5/manpath.5.html#SEARCH_PATH, `"bin/../share/man"` should be a valid search path, so we find the manpage directory in this way.

Reply:
We might also consider adding `UV_TOOL_MAN_DIR` and placing it first.